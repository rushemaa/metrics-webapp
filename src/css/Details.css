.types {
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.type {
  list-style: none;
  width: 100%;
  display: flex;
  color: #fff;
  justify-content: space-between;
}

.type:hover {
  cursor: pointer;
}

.type:nth-child(odd) {
  background-color: #f04c8c;
}

.type:nth-child(even) {
  background-color: #cc3c74;
}

.typeName {
  padding-left: 5px;
}

.typeName::first-letter {
  text-transform: uppercase;
}

.typeBre {
  display: flex;
  align-items: center;
  gap: 20px;
  padding-right: 10px;
}
