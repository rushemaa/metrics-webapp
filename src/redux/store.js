import { configureStore } from '@reduxjs/toolkit';
import texasReducer from './texas/texasSlice';
import breweriesReducer from './texas/breweriesSlice';

const store = configureStore({
  reducer: {
    texas: texasReducer,
    breweries: breweriesReducer,
  },
});

export default store;
