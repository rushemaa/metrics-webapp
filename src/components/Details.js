import { useLocation } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import ArrowCircleRightOutlinedIcon from '@mui/icons-material/ArrowCircleRightOutlined';
import { fetchData } from '../redux/texas/breweriesSlice';
import NavDetails from './NavDetails';
import '../css/Details.css';
import texas from '../assets/texas.svg';

export default function Details() {
  const { state } = useLocation();
  const { breweriesData, loading } = useSelector((bre) => bre.breweries);
  const dispatch = useDispatch();
  useEffect(() => {
    const perPage = state.breweries;
    const link = `by_city=${encodeURIComponent(
      state.city,
    )}&per_page=${perPage}`;
    dispatch(fetchData(link));
  }, [dispatch, state.city, state.breweries]);

  const newBreweries = [];
  breweriesData.forEach((brewery) => {
    const exist = newBreweries.filter(
      (element) => element.type === brewery.brewery_type,
    );
    if (exist.length === 0) {
      const appearance = breweriesData.filter(
        (element) => element.brewery_type === brewery.brewery_type,
      );
      newBreweries.push({
        type: brewery.brewery_type,
        breweries: appearance.length,
      });
    }
  });

  return (
    <>
      <NavDetails />
      <main style={{ backgroundColor: '#F558A1', height: '100vh' }}>
        <section className="main-header">
          <div className="tx-map">
            <img src={texas} className="tx-map-img" alt="Texas map" />
          </div>
          <div className="h-details">
            <div style={{ right: '20px' }}>
              <h2>{state.city}</h2>
              {loading ? (
                <p>Loading...</p>
              ) : (
                <p>
                  {breweriesData.length}
                  {' '}
                  Breweries
                </p>
              )}
            </div>
          </div>
        </section>

        <h4 className="h-status">BREWERIES TYPES BREAKDOWN - 2023</h4>
        {loading ? (
          <p>Loading...</p>
        ) : (
          <ul className="types">
            {newBreweries.map((element) => (
              <li className="type" key={element.type}>
                <p className="typeName">{element.type}</p>
                <div className="typeBre">
                  <p>
                    {element.breweries}
                    {' '}
                    Breweries
                  </p>
                  <ArrowCircleRightOutlinedIcon />
                </div>
              </li>
            ))}
          </ul>
        )}
      </main>
    </>
  );
}
