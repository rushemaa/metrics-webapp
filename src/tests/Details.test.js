import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Details from '../components/Details';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    state: {
      city: 'Austin',
      breweries: '26',
    },
  }),
}));

test('Check if details page renders correctly', () => {
  const tree = render(
    <Provider store={store}>
      <Router>
        <Details />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
