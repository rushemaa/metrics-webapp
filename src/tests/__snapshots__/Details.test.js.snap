// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if details page renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav>
          <div
            class="logo"
          >
            <a
              href="/"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium back-icon css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                />
              </svg>
            </a>
          </div>
          <p>
            City Breweries
          </p>
          <div
            class="navIcons"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icons css-i4bv87-MuiSvgIcon-root"
              data-testid="MicIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icons css-i4bv87-MuiSvgIcon-root"
              data-testid="SettingsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
          </div>
        </nav>
      </header>
      <main
        style="background-color: rgb(245, 88, 161); height: 100vh;"
      >
        <section
          class="main-header"
        >
          <div
            class="tx-map"
          >
            <img
              alt="Texas map"
              class="tx-map-img"
              src="texas.svg"
            />
          </div>
          <div
            class="h-details"
          >
            <div
              style="right: 20px;"
            >
              <h2>
                Austin
              </h2>
              <p>
                Loading...
              </p>
            </div>
          </div>
        </section>
        <h4
          class="h-status"
        >
          BREWERIES TYPES BREAKDOWN - 2023
        </h4>
        <p>
          Loading...
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav>
        <div
          class="logo"
        >
          <a
            href="/"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium back-icon css-i4bv87-MuiSvgIcon-root"
              data-testid="KeyboardArrowLeftIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 16.59 10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
              />
            </svg>
          </a>
        </div>
        <p>
          City Breweries
        </p>
        <div
          class="navIcons"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icons css-i4bv87-MuiSvgIcon-root"
            data-testid="MicIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icons css-i4bv87-MuiSvgIcon-root"
            data-testid="SettingsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </div>
      </nav>
    </header>
    <main
      style="background-color: rgb(245, 88, 161); height: 100vh;"
    >
      <section
        class="main-header"
      >
        <div
          class="tx-map"
        >
          <img
            alt="Texas map"
            class="tx-map-img"
            src="texas.svg"
          />
        </div>
        <div
          class="h-details"
        >
          <div
            style="right: 20px;"
          >
            <h2>
              Austin
            </h2>
            <p>
              Loading...
            </p>
          </div>
        </div>
      </section>
      <h4
        class="h-status"
      >
        BREWERIES TYPES BREAKDOWN - 2023
      </h4>
      <p>
        Loading...
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
